ui/packages/catalog-ui-search/src/main/webapp/react-component/inputs/enum-input/enum-input.js

```diff
@@ -13,37 +13,89 @@
  *
  **/
 import React, { useState } from 'react'
+
 import Dropdown from '../../dropdown'
-import { Menu, MenuItem } from '../../menu'
+import { Menu, MenuItemDisabled } from '../../menu'
 import TextField from '../../text-field'
 import styled from 'styled-components'
 import { getFilteredSuggestions, inputMatchesSuggestions } from './enumHelper'
 import PropTypes from 'prop-types'
-
+const sources = require('../../../component/singletons/sources-instance')
 const TextWrapper = styled.div`
   padding: ${({ theme }) => theme.minimumSpacing};
 `
-
 const EnumMenuItem = props => (
-  <MenuItem {...props} style={{ paddingLeft: '1.5rem' }} />
+  <MenuItemDisabled {...props} style={{ paddingLeft: '1.5rem' }} />
 )
 
+const UnsupportedAttribute = styled.div`
```

Comment:
I could be wrong but I think `EnumInput` was intended to be a generic dropdown component (plus a filter capability) - it shouldn't be aware of sources or attributes. Can we move this logic to a higher level component?

Reply:
I agree just some thoughts I have about that
- I'm not sure how "high" we want to go. The `settingsModel` is originally passed from the query-adanced.view.js and gets passed through several other filter components until it reaches filter.js -> filter-attribute.js -> enum-input.js
- so where ever this logic is moved to I will still need to pass something as a prop instead of `settingsModel` to the enum-input since that's the level that renders each attribute on the dropdown  in order to disable/ enable each attribute.

Reply:
Is there a level where the list of all attributes are passed down? You could filter out attributes from there. If you need to preserve all the attributes, you can instead pass down a list of enabled attributes, along with a list of disabled attributes.

Reply:
Pushed the logic that computes the SupportedAttributes list to `filter.js` so that it passes down the supportedAttributes list to `enum-input.js`.  `enum-input` now just compares wether or not the current attribute that will be rendered is in the list of supportedAttributes